mlserver/rest/endpoints.py

```diff
@@ -93,19 +94,58 @@ async def infer(
         model_name: str,
         model_version: Optional[str] = None,
     ) -> InferenceResponse:
+
         request_headers = dict(raw_request.headers)
         insert_headers(payload, request_headers)
 
         inference_response = await self._data_plane.infer(
             payload, model_name, model_version
         )
-
         response_headers = extract_headers(inference_response)
+
         if response_headers:
             raw_response.headers.update(response_headers)
 
         return inference_response
 
+    async def infer_stream(
+        self,
+        raw_request: Request,
+        raw_response: Response,
+        payload: InferenceRequest,
+        model_name: str,
+        model_version: Optional[str] = None,
+    ) -> StreamingResponse:
+
+        request_headers = dict(raw_request.headers)
+        insert_headers(payload, request_headers)
+
+        async def payloads_async_iter(
+            payload: InferenceRequest,
+        ) -> AsyncIterator[InferenceRequest]:
+            yield payload
+
+        payloads = payloads_async_iter(payload)
+        infer_stream = self._data_plane.infer_stream(
+            payloads, model_name, model_version
+        )
+
+        sse_stream = _as_sse(infer_stream)
+        return StreamingResponse(content=sse_stream, media_type="text/event-stream")
+
+
+async def _as_sse(
+    infer_stream: AsyncIterator[InferenceResponse],
+) -> AsyncIterator[bytes]:
+    """
+    Helper to convert all the responses coming out of a generator to a
+    Server-Sent Event object.
+    """
+    async for inference_response in infer_stream:
+        # TODO: How should we send headers back?
+        # response_headers = extract_headers(inference_response)
```

Comment:
what are the kind of headers we usually send back in the response of `infer`?

Reply:
See link [here](https://github.com/SeldonIO/MLServer/blob/c57a6ce5352625d4ff67569053b43c1a9c360100/mlserver/utils.py#L94-L104)